cmd/device-hub/main.go

```diff
@@ -53,9 +58,43 @@ func (o *config) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&o.KeyFile, "tls-key-file", o.KeyFile, "client key file")
 	fs.StringVar(&o.DataDir, "data-dir", o.DataDir, "path to db folder, defaults to current directory")
 	fs.StringVar(&o.DataImpl, "data-impl", o.DataImpl, "datastore to use, valid values are 'boltdb' or 'filestore', defaults to boltdb")
-	fs.BoolVar(&o.LogFile, "log-file", o.LogFile, "enable log to file")
+	fs.BoolVarP(&o.LogFile, "log-file", "l", o.LogFile, "enable log to file")
 	fs.StringVar(&o.LogPath, "log-path", o.LogPath, "path to log file, defaults to ./device-hub.log")
 	fs.BoolVar(&o.Syslog, "log-syslog", o.Syslog, "enable log to local SYSLOG")
+	fs.BoolVarP(&o.ConfigFile, "config-file", "c", o.ConfigFile, "enable config file overriding flags and env vars")
+	fs.StringVar(&o.ConfigPath, "config-path", o.ConfigPath, "path to config file, defaults to ./config.yaml")
+	fs.Parse(os.Args)
+}
+
+func (o *config) AddConfigFile(cf string) {
+	var readFields interface{}
```

Comment:
Does making `readFields` a `map[string]interface{}` make the rest of this function require less type casting?